### DISCUSSION TITLE
nameof() for Method argument name

### DISCUSSION THREAD

**Discussion by notour** [dotnet/csharplang#7117] [General]
I'm frequently working with dependency injection tools, reflection, ... and for one way to inject a specific value for a parameters is through the argument name.

Example :
```
class ServiceImpl : IService
{
    public ServiceImpl (Guid parentId, IOtherService otherService)
    {
    }
}
```

`service.Resolve<IService>(NameArg.Create("parentId", parent.Id));`

Actually if some dev change the parameter name "parentId" it will cause a bug.

It could be useful to use the power of **nameof()**.
for example nameof(ServiceImpl.ctor(Guid parentId, ...)) it will return at compile time a string "parentId"

OR fix a variable name from a constant, may be through attribute like :

```
class ServiceImpl : IService
{
   public const string ParentIParamName = "parentId";

    public ServiceImpl ([ConstName(ParentIParamName)] Guid parentId, IOtherService otherService)
    {
    }
}
```

**Comment by RikkiGibson** [comment]
This proposes a language change, so moving to csharplang.

**Reply by WhitWaldo** [reply]
Since the values are constant and known ahead of time, could this just be something you create a custom attribute for to specify the const name and then use a source generator to assign it all the way through at build time?
